### PR TITLE
Revert "refactor: move commands away from recipe (#2542)"

### PR DESCRIPTION
**Pull request by olafurpg**
The agent tests started failing after merging the PR #2542 when you re-record the HTTP requests. The tests were passing in replay mode for some reason, and we suspect it's due to a buggy merge in `recording.har`. When you `rm -rf agent/recordings` and record from scratch then the tests fail.

To prevent this from happening again, we're planning to provide scripts to cleanly update the recordings.

## Test plan

Green CI.

### CODE REVIEW

**Review by pkukielka — APPROVED** [review]
LGTM

File: agent/src/index.test.ts [hunk 1]
```diff
@@ -234,9 +234,9 @@ describe('Agent', () => {
             `
           {
             "contextFiles": [],
-            "displayText": " Hello! I don't have any selected code from /src/animal.ts to use. If you provide me with some selected code snippets from that file, I'd be happy to incorporate them into my responses.",
```
Comment (olafurpg) [inline agent/src/index.test.ts:237]: @abeatrix This was a regression from your PR. @pkukielka also has one rate-limiting regression that we discovered from your PR, he will contribute a test to reproduce that regressions. Happy to help you run these tests and troubleshoot the root cause.

**Comment by pkukielka** [comment]
FYI it was failing with:

```
- Expected
+ Received

  {
    "contextFiles": [],
-   "displayText": " Hello!",
+   "displayText": " Hello! I don't have any selected code from /src/animal.ts to use. If you provide me with some selected code snippets from that file, I'd be happy to incorporate them into my responses.",
    "speaker": "assistant",
-   "text": " Hello!",
+   "text": " Hello! I don't have any selected code from /src/animal.ts to use. If you provide me with some selected code snippets from that file, I'd be happy to incorporate them into my responses.",
  }
```

It's also breaking on rate limit tests I just added on another PR (instead of `RateLimitError` it returns empty message).
I will try to merge my PR in the meantime so you would be able to see the issue just by rebasing on top of the main.

**Review by DanTup — APPROVED** [review]

**Comment by abeatrix** [comment]
> FYI when running tests in recording mode  it was failing with:
> 
> 
> 
> ```
> 
> - Expected
> 
> + Received
> 
> 
> 
>   {
> 
>     "contextFiles": [],
> 
> -   "displayText": " Hello!",
> 
> +   "displayText": " Hello! I don't have any selected code from /src/animal.ts to use. If you provide me with some selected code snippets from that file, I'd be happy to incorporate them into my responses.",
> 
>     "speaker": "assistant",
> 
> -   "text": " Hello!",
> 
> +   "text": " Hello! I don't have any selected code from /src/animal.ts to use. If you provide me with some selected code snippets from that file, I'd be happy to incorporate them into my responses.",
> 
>   }
> 
> ```
> 
> 
> 
> It's also breaking on rate limit tests I just added on another PR (instead of `RateLimitError` it returns empty message).
> 
> I will try to merge my PR in the meantime so you would be able to see the issue just by rebasing on top of the main.

@olafurpg @pkukielka what's the issue with the agent test?

i think the correct response should be " Hello! I don't have any selected code from /src/animal.ts to use. If you provide me with some selected code snippets from that file, I'd be happy to incorporate them into my responses." instead of " Hello!", shouldn't it? 🤔

**Comment by olafurpg** [comment]
@abeatrix the current behavior of chat is that Cody replies back with "Hello!" when you post "Hello"

CleanShot 2024-01-05 at 15 05 13@2x


This PR introduced a behavioral change that made Cody complain about missing context, which has been a recurring complaint from customers about Cody complaining about missing context. Also, the title of the PR indicated it was a refactoring, which should mean that the behavior should be unchanged. If the behavioral change is expected then we should make this more clear in the PR title and description.

@pkukielka also discovered another behavioral change related to rate limits, he will add a test case to reproduce this change.
